### PR TITLE
Add tests for upgrade scripts

### PR DESCRIPTION
**Pull request by sumo-drosiek**
###### Description

Add tests for upgrade scripts

###### Testing performed

- [ ] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

**Comment by sumo-drosiek** [comment]
Should I skip log comparison maybe?
@perk-sumo

**Comment by perk** [comment]
`log comparison` is a very nice feature, let's keep it :)

File: tests/upgrade_script/static/falco_disabled.log [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+[INFO]    falco will be disabled. Change "falco.enabled" to "true" if you want to enable it
+
+[INFO]    Replacing tail-db/tail-containers-state.db to tail-db/tail-containers-state-sumo.db
```
Comment (perk-sumo) [inline tests/upgrade_script/static/falco_disabled.log:4]: Hm, that doesn't seem right 🤔 Why we do something with fluentbit when only falco has been changed?

File: tests/upgrade_script/static/basic.log [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
+[INFO]    falco will be enabled. Change "falco.enabled" to "false" if you want to disable it (default for 1.0)
+
+[INFO]    Replacing tail-db/tail-containers-state.db to tail-db/tail-containers-state-sumo.db
```
Comment (perk-sumo) [inline tests/upgrade_script/static/basic.log:4]: Hm I'm not sure we should output anything about fluentbit if there were no changes for it 🤔

File: tests/upgrade_script/static/prometheus_service_monitors.log [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
+[INFO]    falco will be enabled. Change "falco.enabled" to "false" if you want to disable it (default for 1.0)
```
Comment (perk-sumo) [inline tests/upgrade_script/static/prometheus_service_monitors.log:2]: This is not true, falco will be disabled in such case.

File: deploy/helm/sumologic/upgrade-1.0.0.sh [hunk 4]
```diff
@@ -396,25 +396,30 @@ function check_user_image() {
...
+  warning 'Please ensure that new fluent-bit configuration is correct'
 
-  sed 's?tail-db/tail-containers-state.db?tail-db/tail-containers-state-sumo.db?g' ${TEMP_FILE} | \
-  sed "s/'{}'/{}/g" > new_values.yaml
+  sed -i '' 's?tail-db/tail-containers-state.db?tail-db/tail-containers-state-sumo.db?g' ${TEMP_FILE}
 }
 
-function remove_temp_file() {
-  rm ${TEMP_FILE}
+function fix_yq() {
```
Comment (perk-sumo) [inline deploy/helm/sumologic/upgrade-1.0.0.sh:412]: 👍

File: tests/upgrade_script/static/key_mapping_empty.output.yaml [hunk 5]
```diff
@@ -0,0 +1 @@
+{fluentd: {}, sumologic: {}}
```
Comment (perk-sumo) [inline tests/upgrade_script/static/key_mapping_empty.output.yaml:1]: @sumo-drosiek you sure that's correct? Doesn't look like 🤔

Comment (sumo-drosiek) [reply tests/upgrade_script/static/key_mapping_empty.output.yaml:1]: Yes, it's valid yaml. Do you want to say that we should check all keys and remove them if they are empty objects?
What if customer have empty object in it's yaml file?

Comment (perk-sumo) [reply tests/upgrade_script/static/key_mapping_empty.output.yaml:1]: You are right, thank you :)

File: tests/upgrade_script/static/key_mapping_empty.log [hunk 6]
```diff
@@ -0,0 +1,22 @@
+[INFO]    Mapping deployment into:
+[INFO]    Removing deployment
+[INFO]    Removing eventsDeployment
+[INFO]    Mapping fluentd.autoscaling into:
```
Comment (perk-sumo) [inline tests/upgrade_script/static/key_mapping_empty.log:4]: What happened to this `autoscaling`?

Comment (sumo-drosiek) [reply tests/upgrade_script/static/key_mapping_empty.log:4]: It should be object instead of string, but it's not the case of this test

**Pull request merged**